cmd/podman/system/service.go

```diff
@@ -131,20 +131,17 @@ func resolveAPIURI(_url []string) (string, error) {
 		if srvArgs.Varlink {
 			socketName = "io.podman"
 		}
-		socketDir := filepath.Join(xdg, "podman", socketName)
-		if _, err := os.Stat(filepath.Dir(socketDir)); err != nil {
-			if os.IsNotExist(err) {
-				if err := os.Mkdir(filepath.Dir(socketDir), 0755); err != nil {
-					return "", err
-				}
-			} else {
-				return "", err
-			}
+		socketPath := filepath.Join(xdg, "podman", socketName)
+		if err := os.MkdirAll(filepath.Dir(socketPath), 0755); err != nil {
```

Comment:
Should this be more restrictive than 755? We may want to 700 for security reasons

Reply:
700 would make sense. Others shouldn't access it.